src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -336,24 +336,43 @@ await Task.WhenAll(
             }
         }
 
-        private void ReportFileChanges(IReadOnlyList<FileItem> fileItems)
+        private void ReportFileChanges(IReadOnlyList<ChangedFile> changedFiles)
         {
-            Report(added: true);
-            Report(added: false);
+            Report(kind: ChangeKind.Add);
+            Report(kind: ChangeKind.Update);
+            Report(kind: ChangeKind.Delete);
 
-            void Report(bool added)
+            void Report(ChangeKind kind)
             {
-                var items = fileItems.Where(item => item.IsNewFile == added).ToArray();
+                var items = changedFiles.Where(item => item.Change == kind).ToArray();
                 if (items is not [])
                 {
-                    Context.Reporter.Output(GetMessage(items, added));
+                    Context.Reporter.Output(GetMessage(items, kind));
                 }
             }
 
-            string GetMessage(IReadOnlyList<FileItem> items, bool added)
-                => items is [var item]
-                    ? (added ? "File added: " : "File changed: ") + GetRelativeFilePath(item.FilePath)
-                    : (added ? "Files added: " : "Files changed: ") + string.Join(", ", items.Select(f => GetRelativeFilePath(f.FilePath)));
+            string GetMessage(IReadOnlyList<ChangedFile> items, ChangeKind kind)
+                => items is [{Item: var item }]
+                    ? GetSingularMessage(kind) + ": " + GetRelativeFilePath(item.FilePath)
+                    : GetPlurarMessage(kind) + ": " + string.Join(", ", items.Select(f => GetRelativeFilePath(f.Item.FilePath)));
+
+            static string GetSingularMessage(ChangeKind kind)
+                => kind switch
+                {
+                    ChangeKind.Update => "File updated",
+                    ChangeKind.Add => "File added",
+                    ChangeKind.Delete => "File deleted",
+                    _ => throw new InvalidOperationException()
+                };
+
+            static string GetPlurarMessage(ChangeKind kind)
+                => kind switch
+                {
+                    ChangeKind.Update => "Files updated",
```

Comment:
Should these "File updated", "File added" be translatable strings?

Reply:
Eventually, yes. Watch has many non-translatable strings. We should do a pass to move all strings to resources.